power_grid_model_c/power_grid_model/include/power_grid_model/container.hpp

```diff
@@ -110,6 +110,15 @@ class Container<RetrievableTypes<GettableTypes...>, StorageableTypes...> {
         assert(is_base<Gettable>[idx_2d.group]);
         return (this->*(func_arr[idx_2d.group]))(idx_2d.pos);
     }
+    // get id by idx
+    ID get_id_by_idx(Idx2D idx_2d) const {
+        for (auto const& [key, value] : map_) {
+            if (value == idx_2d) {
+                return key;
+            }
+        }
```

Comment:
Is this inefficient? We do a search every time just to find the id.

Reply:
And are there realistic scenarios that the `Idx2D` cannot be found? This should be a program error right?

Reply:
Problem is there are two types if `Idx2D` (https://github.com/PowerGridModel/power-grid-model/issues/798) and it has linear complexity and is only called in one assert in `updata.hpp`.

Reply:
If it's only called in `assert`, maybe wrap it for now to make sure it is not used in release build.